live_data_subscriber/__init__.py

```diff
@@ -60,15 +47,19 @@ def generate_topic_api_client_args_mapping():
             get_area_uuid_and_name_mapping_from_simulation_id(configuration_id, sim_api_domain_name)
         # It is a prerequisite to have unique area names for this to work.
         uuid_name_mapping = {v: k for k, v in name_uuid_mapping.items()}
-
         for device_uuid in live_data_device_uuids:
             device_name = uuid_name_mapping[device_uuid]
-            if device_name in allowed_devices_name_mapping:
-                topic_name = allowed_devices_name_mapping[device_name]
-                topic_api_client_mapping[topic_name].append({
-                    "device_id": device_uuid,
-                    **api_client_args
-                })
-
-    logging.info(f"Connecting to the following MQTT topics/devices {topic_api_client_mapping}")
-    return topic_api_client_mapping
+            if device_name in allowed_device_mapping:
+                api_client_mapping[allowed_device_mapping[device_name]].append(
+                    {"device_id": device_uuid, **api_client_args}
+                )
+    return api_client_mapping
+
+
+def refresh_cn_and_device_list(last_time_checked, api_client_dict,
+                               default_api_client_map):
```

Comment:
Isnt it rather a `allowed_device_mapping` than a `default_api_client_map` ? 
```suggestion
                               allowed_device_mapping):
```